src/adapter/src/catalog.rs

```diff
@@ -19,6 +19,7 @@ use anyhow::bail;
 use chrono::{DateTime, Utc};
 use futures::Future;
 use itertools::Itertools;
+use mz_persist_client::cfg::PersistParameters;
```

Comment:
This import should move to the `mz_` import block below. Looks like that's the case in other files changed here as well!

Reply:
Why isn't there a `clippy` / `rustfmt` extension / rule that we can use for that?

Reply:
I checked that before. [rustfmt has a related option](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#group_imports), but a) it's unstable and b) it doesn't allow specifying a first-party import block. Clippy doesn't have a fitting lint either, I guess it's just not concerned with formatting.

I think our best bet at linting this writing some regex-based script to detect misordering. Unless there is a way to extend rustfmt somehow...

Reply:
done!

I happen to have a different take (which you could probably guess by looking at persist code) which is, given that a huge majority of us use rust-analyzer and it doesn't seem to support this, that it's not really worth the effort and we should use the standard rust grouping `(std::, all other crates, crate::)` instead. however, I absolutely don't mean to pick that off as part of this PR :) or to mess up y'alls existing conventions, so happy to fix them up here. sorry for not being more careful in the first place.

Reply:
Summoning @benesch for his take. Given that enforcing the 4-group separation seems to be non-trivial, maybe we can consider just using the [rustfmt option](https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#group_imports) that @teskje mentioned.

Reply:
Although it goes against my esthetic sense, I would be totally on board with changing our convention to group mz_ crates with external crates! Both for the reason you mentioned, and because we could use (nightly) rustfmt to enforce this style. I'm sure I would get used to it looking strange pretty quickly, and it's definitely better than the current situation where there is one prevalent style, which is not enforceable and therefore not consistently followed.

But yeah, that probably needs to be a separate discussion.